File: ios/Sources/PrivacyScreenPlugin/PrivacyScreenPlugin.swift
import Foundation
import Capacitor

@objc(PrivacyScreenPlugin)
public class PrivacyScreenPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "PrivacyScreenPlugin"
    public let jsName = "PrivacyScreen"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "enable", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "disable", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isEnabled", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getPluginVersion", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = PrivacyScreen()

    override public func load() {
        DispatchQueue.main.async {
            self.implementation.start { [weak self] in
                self?.bridge?.viewController?.view.window
            }
        }
    }

    @objc func enable(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.implementation.setEnabled(true)
            call.resolve([
                "enabled": self.implementation.isEnabled
            ])
        }
    }

    @objc func disable(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.implementation.setEnabled(false)
            call.resolve([
                "enabled": self.implementation.isEnabled
            ])
        }
    }

    @objc func isEnabled(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            call.resolve([
                "enabled": self.implementation.isEnabled
            ])
        }
    }

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve([
            "version": Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "ios"
        ])
    }
}

File: android/src/main/java/app/capgo/privacyscreen/PrivacyScreenPlugin.java
package app.capgo.privacyscreen;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PrivacyScreen")
public class PrivacyScreenPlugin extends Plugin {

    private PrivacyScreen implementation;

    @Override
    public void load() {
        implementation = new PrivacyScreen(getActivity());
        implementation.enable();
    }

    @PluginMethod
    public void enable(final PluginCall call) {
        implementation.enable();
        call.resolve(status());
    }

    @PluginMethod
    public void disable(final PluginCall call) {
        implementation.disable();
        call.resolve(status());
    }

    @PluginMethod
    public void isEnabled(final PluginCall call) {
        call.resolve(status());
    }

    @PluginMethod
    public void getPluginVersion(final PluginCall call) {
        final JSObject ret = new JSObject();
        ret.put("version", "android");
        call.resolve(ret);
    }

    private JSObject status() {
        final JSObject ret = new JSObject();
        ret.put("enabled", implementation != null && implementation.isEnabled());
        return ret;
    }
}
